components/scripts/lib/build-scan-parse.sh

```diff
@@ -74,28 +73,48 @@ debug_build_scan_data() {
 
 # shellcheck disable=SC2034 # not all scripts use all of the fetched data
 parse_build_scan_row() {
-  local build_cache_metrics_only="$1"
-  local build_scan_row="$2"
+  local build_scan_row="$1"
 
   local run_num
 
   while IFS=, read -r run_num field_1 field_2 field_3 field_4 field_5 field_6 field_7 field_8 field_9 field_10 field_11 field_12 field_13 field_14 field_15 field_16 field_17 field_18 field_19 field_20 field_21; do
     debug "Build Scan $field_4 is for build $run_num"
-    project_names[run_num]="$field_1"
-    build_scan_ids[run_num]="$field_4"
-
-    if [[ "$build_cache_metrics_only" != 'build_cache_metrics_only' ]]; then
-      base_urls[run_num]="$field_2"
-      build_scan_urls[run_num]="$field_3"
-      git_repos[run_num]="$field_5"
-      git_branches[run_num]="$field_6"
-      git_commit_ids[run_num]="$field_7"
+
+    project_names[run_num]="${field_1}"
+
+    if [ -z "${base_urls[run_num]}" ]; then
+      base_urls[run_num]="${field_2}"
+    fi
+
+    if [ -z "${build_scan_urls[run_num]}" ]; then
+      build_scan_urls[run_num]="${field_3}"
+    fi
+
+    build_scan_ids[run_num]="${field_4}"
+
+    if [ -z "${git_repos[run_num]}" ]; then
+      git_repos[run_num]="${field_5}"
+    fi
+
+    if [ -z "${git_branches[run_num]}" ]; then
+      git_branches[run_num]="${field_6}"
+    fi
+
+    if [ -z "${git_commit_ids[run_num]}" ]; then
+      git_commit_ids[run_num]="${field_7}"
+    fi
+
+    if [ -z "${requested_tasks[run_num]}" ]; then
       requested_tasks[run_num]="$(remove_clean_task "${field_8}")"
-      build_outcomes[run_num]="$field_9"
-      remote_build_cache_urls[run_num]="${field_10}"
-      remote_build_cache_shards[run_num]="${field_11}"
     fi
 
+    if [ -z "${build_outcomes[run_num]}" ]; then
+      build_outcomes[run_num]="${field_9}"
+    fi
+
+    remote_build_cache_urls[run_num]="${field_10}"
+    remote_build_cache_shards[run_num]="${field_11}"
+
```

Comment:
What is field_20 (it seems not to be read further down) ?

Reply:
Good question. Field 20 is "Build time". It is no longer used as we return "Initial build time" and "Instant build time" in the new build time metrics part of the response. It probably should have been removed as part of #383.

Reply:
I see. Thanks.